Documentation version 2 of the translation system (#899)
danskernesdigitalebibliotek/dpl-cms: docs/architecture/adr-011-configuration-translation-system.md

# Architecture Decision Record: Configuration translation system

## Context

> The translation system desribed in [adr-009-translation-system](./adr-009-translation-system.md) handles solely the translations added through Drupals traditional translation handling.

Reviewer: ```suggestion
The translation system described in [adr-009-translation-system](./adr-009-translation-system.md) handles solely the translations added through Drupals traditional translation handling.
```

But there was a wish for being able to translate configuration related strings as well: Titles, labels, descriptions etc. in the admin area.

We needed to find a way to handle translation of that kind as well.

## Decision

We went for a solution where we activated the Configuration Translation Drupal core module and added the Configuration Translation PO contrib module.

> And we added a range of custom drush commands to handle the varios configuration translation tasks.

Reviewer: ```suggestion
And we added a range of custom drush commands to handle the various configuration translation tasks.
```

## Consequences

> By sticking to the handling of PO files in configuration handling that we are already using in our genral translation handling, we can keep the current Github workflows with some alterations.

Reviewer: ```suggestion
By sticking to the handling of PO files in configuration handling that we are already using in our general translation handling, we can keep the current Github workflows with some alterations.
```

> By sticking to the handling of PO files in configuration handling that we are already using in our general translation handling, we can keep the current Github workflows with some alterations.

Reviewer 2: Consider mentioning that using this approach handling translations of configuration on local sites is still as difficult as before. Translation of configuration texts cannot be found in the standard UI strings translation list.

### Alterations to former translation workflow

> With the config translation PO files added we tried to uncover if PoEditor was able to handle two PO files simultaniously in both import and export context.

Reviewer: ```suggestion
With the config translation PO files added we tried to uncover if PoEditor was able to handle two PO files simultaneously in both import and export context.
```

> With the config translation PO files added we tried to uncover if PoEditor was able to handle two PO files simultaniously in both import and export context.

Reviewer 3: ```suggestion
With the config translation PO files added we tried to uncover if POEditor was able to handle two PO files simultaniously in both import and export context.
```

It could not.

But we still needed, in Drupal, to be able to import two different files: One for general translations and one for configuration translations.

> We came up with the idea that we could merge the two files going when importing into Poeditor and split it again when exporting form PoEditor.

Reviewer: ```suggestion
We came up with the idea that we could merge the two files going when importing into Poeditor and split it again when exporting from PoEditor.
```

> We came up with the idea that we could merge the two files going when importing into Poeditor and split it again when exporting form PoEditor.

Reviewer 3: ```suggestion
We came up with the idea that we could merge the two files when importing into POEditor and split it again when exporting from POEditor.
```

> We came up with the idea that we could merge the two files going when importing into POEditor and split it again when exporting from POEditor.

Reviewer 2: Consider elaborating with how we handle splitting in practice.

Author: It is documented in the other file.

We tried it out and it worked so that was the solution we ended up with.

## Alternatives considered

### A hack

> We could Keep the machine names of the config in english but writing the titles, lables, descriptions in danish.

Reviewer: ```suggestion
We could Keep the machine names of the config in english but writing the titles, labels, descriptions in danish.
```

> We could Keep the machine names of the config in english but writing the titles, labels, descriptions in danish.

Reviewer 2: ```suggestion
We could keep the machine names of the config in English but write the titles, labels, descriptions in Danish.
```

> But that would have following bad consequences:

Reviewer 3: ```suggestion
But that would have the following bad consequences:
```

1. The administrators would have to find all the texts in various, not obvious, places in the admin area.

> 2. It would differ from the genral translation routine which is confusing

Reviewer: ```suggestion
2. It would differ from the general translation routine which is confusing
```

> 2. It would differ from the genral translation routine which is confusing

Reviewer 3: ```suggestion
2. It would differ from the general translation routine which is confusing.
```

> 3. We would not be able to hande multiple languages for the cofiguration translations

Reviewer: ```suggestion
3. We would not be able to handle multiple languages for the configuration translations
```

> 3. We would not be able to hande multiple languages for the cofiguration translations

Reviewer 3: ```suggestion
3. We would not be able to handle multiple languages for the configuration translations.
```
